### PR TITLE
[zh]translate kubernetes-api/workload-resources/controller-revision-v1.md into Chinese

### PR DESCRIPTION
**Pull request by twilight0620**
Fixes: https://github.com/kubernetes/website/issues/33644

Do not have chinese docs: https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/controller-revision-v1/

so need to translatekubernetes-api/workload-resources/controller-revision-v1.md into Chinese

### CODE REVIEW

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 1]
```diff
@@ -0,0 +1,1065 @@
...
+- **kind**: ControllerRevision
+
+<!--
+- **metadata** (<a href="{{< ref "../common-definitions/object-meta#ObjectMeta" >}}">ObjectMeta</a>)
+
+  Standard object's metadata. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+-->
+- **metadata** (<a href="{{< ref "../common-definitions/object-meta#ObjectMeta" >}}">ObjectMeta</a>)
+
+  标准对象的元数据。
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:74]:
```suggestion
  标准的对象元数据。
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 2]
```diff
@@ -0,0 +1,1065 @@
...
+  更多信息：https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+
+<!--
+- **revision** (int64), required
+
+  Revision indicates the revision of the state represented by Data.
+-->
+- **revision** (int64)，必需
+
+  Revision 表示 Data 表示的状态的修订。
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:84]:
```suggestion
  revision 表示 data 表示的状态的修订。
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 3]
```diff
@@ -0,0 +1,1065 @@
...
+
+<!--
+- **data** (RawExtension)
+
+  Data is the serialized representation of the state.
+-->
+- **data** (RawExtension)
+
+  Data 是状态的序列化表示。
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:93]:
```suggestion
  data 是状态的序列化表示。
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 4]
```diff
@@ -0,0 +1,1065 @@
...
+  
+  <!--
+  <a name="RawExtension"></a>
+    *RawExtension is used to hold extensions in external versions.
+  -->
+  <a name="RawExtension"></a>
+  *RawExtension 用于保存外部版本中的扩展。
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:100]:
```suggestion
  *RawExtension 用于以外部版本来保存扩展数据。
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 5]
```diff
@@ -0,0 +1,1065 @@
...
+  
+  <!--
+  To use this, make a field which has RawExtension as its type in your external, versioned struct, and Object in your internal struct. You also need to register your various plugin types.
+  -->
+  要使用它，请在外部、版本化结构和内部结构中创建一个类型为 RawExtension 的字段。
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:105]:
```suggestion
  要使用它，请生成一个字段，在外部、版本化结构中以 RawExtension 作为其类型，在内部结构中以 Object 作为其类型。
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 6]
```diff
@@ -0,0 +1,1065 @@
...
+    } type PluginA struct {
+    	AOption string `json:"aOption"`
+    }
+  -->
+  // 内部包：type MyAPIObject struct {
+  	runtime.TypeMeta `json:",inline"`
+  	MyPlugin runtime.Object `json:"myPlugin"`
+  } type PluginA struct {
+  	AOption string `json:"aOption"`
+  }
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:121]: 这里的英文排版有问题，中文翻译时需要校正。下同

````suggestion
```go
  // 内部包：
  type MyAPIObject struct {
  	runtime.TypeMeta `json:",inline"`
  	MyPlugin runtime.Object `json:"myPlugin"`
  }
  type PluginA struct {
  	AOption string `json:"aOption"`
  }
```
````

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 7]
```diff
@@ -0,0 +1,1065 @@
...
+  		"aOption":"foo",
+  	},
+  }
+  
+  <!--
+  So what happens? Decode first uses json or yaml to unmarshal the serialized data into your external MyAPIObject. That causes the raw JSON to be stored, but not unpacked. The next step is to copy (using pkg/conversion) into the internal struct. The runtime package's DefaultScheme has conversion functions installed which will unpack the JSON stored in RawExtension, turning it into the correct object type, and storing it in the Object. (TODO: In the case where the object is of an unknown type, a runtime.Unknown object will be created and stored.)*
+  -->
+  那么会发生什么？
+  解码首先使用 json 或 yaml 将序列化数据解组到你的外部 MyAPIObject 中。
+  这会导致原始 JSON 被存储，但不会被解包。
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:162]:
```suggestion
  这会导致原始 JSON 被存储下来，但不会被解包。
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 8]
```diff
@@ -0,0 +1,1065 @@
...
+  下一步是复制（使用 pkg/conversion）到内部结构中。
+  运行时包的 DefaultScheme 安装了转换函数，它将解压存储在 RawExtension 中的 JSON，
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:164]:
```suggestion
  runtime 包的 DefaultScheme 安装了转换函数，它将解析存储在 RawExtension 中的 JSON，
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 9]
```diff
@@ -0,0 +1,1065 @@
...
+  将其转换为正确的对象类型，并将其存储在 Object.xml 中。
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:165]:
```suggestion
  将其转换为正确的对象类型，并将其存储在 Object 中。
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 10]
```diff
@@ -0,0 +1,1065 @@
...
+  （TODO：如果对象是未知类型，将创建并存储一个 `runtime.Unknown`对象。）*
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:166]:
```suggestion
  （TODO：如果对象是未知类型，将创建并存储一个 `runtime.Unknown`对象。）
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 11]
```diff
@@ -0,0 +1,1065 @@
...
+  更多信息：https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+
+<!--
+- **items** ([]<a href="{{< ref "../workload-resources/controller-revision-v1#ControllerRevision" >}}">ControllerRevision</a>), required
+
+  Items is the list of ControllerRevisions
+-->
+- **items** ([]<a href="{{< ref "../workload-resources/controller-revision-v1#ControllerRevision" >}}">ControllerRevision</a>)，必需
+
+  Items 是 ControllerRevisions 的列表
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:207]:
```suggestion
  items 是 ControllerRevisions 的列表
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 12]
```diff
@@ -0,0 +1,1065 @@
...
+#### 参数
+
+<!--
+- **name** (*in path*): string, required
+
+  name of the ControllerRevision
+-->
+- **name** （**路径参数**）： string，必需
+
+  ControllerRevision 名称
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:244]:
```suggestion
  ControllerRevision 的名称
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 13]
```diff
@@ -0,0 +1,1065 @@
...
+#### Parameters
+-->
+#### 参数
+
+<!--
+- **name** (*in path*): string, required
+
+  name of the ControllerRevision
+-->
+- **name** （**路径参数**）： string，必需
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:242]: 建议删掉多余的空格，不然间隔太大。下同。

```suggestion
- **name** （**路径参数**）：string，必需
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 14]
```diff
@@ -0,0 +1,1065 @@
...
+
+
+200 (<a href="{{< ref "../workload-resources/controller-revision-v1#ControllerRevision" >}}">ControllerRevision</a>): OK
+
+401: Unauthorized
+
+<!--
+### `list` list or watch objects of kind ControllerRevision
+-->
+### `list` 列出或观察 ControllerRevision 种类的对象
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:282]:
```suggestion
### `list` 列出或监视 ControllerRevision 类别的对象
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 15]
```diff
@@ -0,0 +1,1065 @@
...
+#### 响应
+
+200 (<a href="{{< ref "../workload-resources/controller-revision-v1#ControllerRevisionList" >}}">ControllerRevisionList</a>): OK
+
+401: Unauthorized
+
+<!--
+### `list` list or watch objects of kind ControllerRevision
+-->
+### `list` 列出或观察 ControllerRevision 种类的对象
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:413]:
```suggestion
### `list` 列出或监视 ControllerRevision 类别的对象
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 16]
```diff
@@ -0,0 +1,1065 @@
...
+#### 参数
+
+<!--
+- **name** (*in path*): string, required
+
+  name of the ControllerRevision
+-->
+- **name** （**路径参数**）： string，必需
+
+  ControllerRevision 名称
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:649]:
```suggestion
  ControllerRevision 的名称
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 17]
```diff
@@ -0,0 +1,1065 @@
...
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:745]:
```suggestion
  ControllerRevision 的名称
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 18]
```diff
@@ -0,0 +1,1065 @@
...
+#### Parameters
+-->
+#### 参数
+
+<!--
+- **name** (*in path*): string, required
+
+  name of the ControllerRevision
+-->
+- **name** （**路径参数**）： string，必填
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:848]:
```suggestion
- **name** （**路径参数**）：string，必需
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md [hunk 19]
```diff
@@ -0,0 +1,1065 @@
...
+
+  ControllerRevision 名称
```
Comment (tengqm) [inline content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:850]:
```suggestion
  ControllerRevision 的名称
```

File: content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md (hunk 10 above)
Comment (twilight0620) [reply content/zh/docs/reference/kubernetes-api/workload-resources/controller-revision-v1.md:166]: > 

100 行是有个星号起始的，这块应该是结束的星号吧。
*RawExtension 用于以外部版本来保存扩展数据。

**Comment by twilight0620** [comment]
I have modified all the comments, thanks for your review.

**Pull request merged**